build_tools/cmake/easy.sh

```diff
@@ -0,0 +1,291 @@
+  echo "  Debug|Release|RelWithDebInfo|MinSizeRel    Set the CMake build type."
+  echo "  src <path>   Specify IREE source path. Default: inferred from script path."
+  echo "  ndk <path>   Specify Android NDK path. Configure an Android build."
+  echo "  android_abi <abi> Specify Android ABI. Optional. Default: ${android_abi}." 
+  echo "  android_platform <platform> Specify Android platform version. Optional. Default: ${android_platform}."
+  echo "  docs         Build docs."
+  echo "  tracy        Build with Tracy profiler instrumentation."
+  echo "  py           Build Python bindings."
+  echo "  java         Build Java bindings."
+  echo "  tf           Build TensorFlow compiler frontend."
+  echo "  tflite       Build TFLite compiler frontend."
+  echo "  xla          Build XLA compiler frontend."
+  echo
+  echo "Examples:"
+  echo
+  echo "  Configure and build with ASan, CCache, and Tracy. Note how 'build' comes last!"
+  echo "    $(basename "$0") asan ccache tracy build"
+  echo
+  echo "  Configure and build and Android NDK build, passing the NDK path. Also with CCache and Tracy."
+  echo "    $(basename "$0") ndk ~/android-ndk-r21d ccache tracy build"
+  echo
+  echo "  Configure a Debug build, build only iree_tools_iree-translate:"
+  echo "    $(basename "$0") Debug build iree_tools_iree-translate"
+  echo
+  echo "  Configure, but do not build, a Release build with Python bindings and TensorFlow compiler:"
+  echo "    $(basename "$0") Release py tf"
+  echo
+  echo "  Do not configure, just build iree_tools_iree-translate"
+  echo "    $(basename "$0") build iree_tools_iree-translate"
+
+  exit 1
+fi
+
+# Parse command-line args
+args=("$@")
+for (( i=0; i < $#; i++ ))
+do
+  case "${args[i]}" in
+    noclear) arg_noclear=1;;
+    printonly) arg_printonly=1;;
+    ccache) arg_ccache=1;;
+    asan) arg_asan=1;;
+    msan) arg_msan=1;;
+    tsan) arg_tsan=1;;
+    py) arg_py=1;;
+    java) arg_java=1;;
+    tf) arg_tf=1;;
+    tflite) arg_tflite=1;;
+    xla) arg_xla=1;;
+    tracy) arg_tracy=1;;
+    docs) arg_docs=1;;
+    Debug|Release|RelWithDebInfo|MinSizeRel) arg_build_type="${args[i]}";;
+    ndk) arg_ndk="$(realpath -s "${args[$((i+1))]}")"; i=$((i+1));;
+    android_abi) android_abi="${args[$((i+1))]}"; i=$((i+1));;
+    android_platform) android_platform="${args[$((i+1))]}"; i=$((i+1));;
+    src) arg_src="$(realpath -s "${args[$((i+1))]}")"; i=$((i+1));;
+    build)
+      arg_build=1
+      if [[ $i < $(($# - 1)) ]]
+      then
+        arg_targets="${args[@]:$((i+1))}"
+      fi  
+      break;;
+    *) echo "Error: unkown argument ${args[i]}" >&2; exit 1;;
+  esac
+done
+
+if [[ "${arg_printonly}" != 1 && -f "CMakeLists.txt" ]]
+then
+  echo "Error: We seem to be in a source directory. Please cd into a build directory." >&2
+  exit 1
+fi
+
+if [[ -z "${arg_build_type}" ]]
+then
+  # Sane default build type
+  # TODO: upstream this sane default to IREE's CMakeLists
+  arg_build_type=RelWithDebInfo
+fi
```

Comment:
Build type is a generic CMake setting (https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html) and I don't see a compelling reason to make our project special.

Assuming this script is exclusively for developers and not dependent projects/users, having a _developer default_ of `RelWithDebInfo` in this script seems reasonable.

Can you reword the comment here?

Reply:
Who cares about the storage size benefits of omitting debug info? Only users who make IREE binaries *and distribute them to other users*. That's not the majority of IREE users, who will be making IREE binaries for their own use locally, and probably don't mind the storage size cost of IREE debug symbols.

Who cares about the debuggability improvement of having debug info? Everyone does, if only for filing bugs, or for profiling. Even end-users would benefit, and omitting debug info for them is only done as a compromise with download size.

This is a generic argument, not specific to IREE, so RelWithDebInfo really is a saner default than Release. Why would CMake have suboptimal defaults? Because as a build system, so much software is relying on its defaults that it must be very hard to ever change defaults. Not surprising, no need for IREE to be special.